indico/modules/auth/client/js/signup.jsx

```diff
@@ -67,6 +74,34 @@ function Signup({
     await new Promise(() => {});
   };
 
+  const termsAcceptLabel = () => {
+    const tosElement = tosUrl ? (
+      <a href="/tos" target="_blank">
+        {Translate.string('terms of service')}
+      </a>
+    ) : (
+      Translate.string('terms of service')
+    );
+    const ppElement = privacyPolicyUrl ? (
+      <a href="/pp" target="_blank">
+        {Translate.string('privacy policy')}
+      </a>
+    ) : (
+      Translate.string('privacy policy')
+    );
+    if ((tos || tosUrl) && (privacyPolicy || privacyPolicyUrl)) {
+      return (
+        <>
+          {tosElement} {Translate.string('and')} {ppElement}
```

Comment:
please do not build translated strings like this. it doesn't work well depending on the language. it' possible to use `<Translate>` and `<Param>` to build this string without this "mess" (untested, though):

```js
import privacyPolicyURL from 'indico-url:legal.display_privacy';
import tosURL from 'indico-url:legal.display_tos';
```

```jsx
<Translate>
  I have read, understood, and accept the{' '}
  <Param name="tosLink" wrapper={<a href={tosURL()} target="_blank" />}>
    Terms of Service
  </Param>{' '}
  and{' '}
  <Param name="privacyLink" wrapper={<a href={privacyPolicyURL()} target="_blank" />}>
    Privacy Policy
  </Param>{' '}
</Translate>
```

for the other two variants, simply duplicate and adapt this block.

Reply:
Thanks, I wasn't aware of `<Param>`. I found a good way to do this in jinja as well.